### DISCUSSION TITLE
How to define field with argument in ObjectType?

### DISCUSSION THREAD

**Discussion by waptik** [MichalLytek/type-graphql#966] [Q&A]
Hi! I'm new to typegraphql. I'm trying to migrate from SDL to typegraphql.
I have some complex type and need some help wit it:

I have some enum defined as follow
```ts
// enums.ts
...
export enum ImageFormat {
JPG='JPG',
GIF='GIF',
WEBP='WEBP'
}

export enum Role{
USER='USER',
ADMIN='ADMIN',
ROOT='ROOT'
}

registerEnumType(ImageFormat , { name: 'ImageFormat ' });
registerEnumType(Role, { name: 'Role' });

...

// end enums.ts

```

How do i achieve the following?

```gql
type User{
      id: String!
      name: String!
      roles: [Role]!
      isAdmin: Boolean!
      since: DateTime!
      type: String!
      description: String
      username: String
      "User's image"
      image(
        height: Int,
        format: ImageFormat
      ): String
    }

```

I am not able to properly define field argument like `image(height: Int, format:ImageFormat): String`

**Comment by MichalLytek** [comment]
Have you read the docs?
https://typegraphql.com/docs/resolvers.html#field-resolvers

**Reply by waptik** [reply]
I sure did. But it seem i misunderstood some stuffs over there.
What i did was a follow:
```ts
// User.ts
@ObjectType()
export class User {
...
@Field()
image(@Arg("height") height: number, @Arg("format") format: ImageFormat): string // was wrong
}

```
I did the above so that i could create a type resolver based on it, but i got the following error `Unable to resolve signature of property decorator when called as an expression.ts(1240)`

But i failed to notice this line:
```md
Note that if a field name of a field resolver doesn't exist in the resolver object type, it will create a field in the schema with this name. This feature is useful when the field is purely calculable (eg. averageRating from ratings array) and to avoid polluting the class signature.
```
So meaning i just have to omit `image()` from `User` and move it to `UserResolver` as a `FieldResolver`. Correct me i am wrong.
